packages/react/src/components/OverflowMenu/index.jsx

```diff
@@ -1,18 +1,604 @@
+     */
+    renderIcon: PropTypes.oneOfType([PropTypes.func, PropTypes.object]),
+
+    /**
+     * Specify a CSS selector that matches the DOM element that should
+     * be focused when the OverflowMenu opens
+     */
+    selectorPrimaryFocus: PropTypes.string,
+
+    /**
+     * Specify the size of the OverflowMenu. Currently supports either `sm`, 'md' (default) or 'lg` as an option.
+     */
+    size: PropTypes.oneOf(['sm', 'md', 'lg', 'xl']),
+
+    /**
+     * Test id for the trigger component
+     */
+    'data-testid': PropTypes.string.isRequired,
+
+    /**
+     * Specify the alignment of the tooltip to the icon-only button.
+     * Can be one of: start, center, or end.
+     */
+    tooltipAlignment: PropTypes.oneOf(['start', 'center', 'end']),
+
+    /**
+     * Specify the direction of the tooltip for icon-only buttons.
+     * Can be either top, right, bottom, or left.
+     */
+    tooltipPosition: PropTypes.oneOf(['top', 'right', 'bottom', 'left']),
+
+    /**
+     * Language direction
+     */
+    langDir: PropTypes.oneOf(['ltr', 'rtl']),
+  };
+
+  // eslint-disable-next-line react/sort-comp
+  static contextType = PrefixContext;
+
+  static defaultProps = {
+    ariaLabel: 'open and close list of options',
+    children: [],
+    className: '',
+    direction: DIRECTION_BOTTOM,
+    flipped: false,
+    focusTrap: true,
+    iconClass: '',
+    open: false,
+    menuOffset: getMenuOffset,
+    menuOffsetFlip: getMenuOffset,
+    menuOptionsClass: '',
+    light: false,
+    renderIcon: OverflowMenuVertical20,
+    selectorPrimaryFocus: '[data-overflow-menu-primary-focus]',
+    tooltipAlignment: 'center',
+    tooltipPosition: 'top',
+    size: 'md',
+    langDir: 'ltr',
+  };
+
+  /**
+   * The handle of `onfocusin` or `focus` event handler.
+   * @private
+   */
+  // eslint-disable-next-line react/sort-comp
+  _hFocusIn = null;
+
+  /**
+   * The timeout handle for handling `blur` event.
+   * @private
+   */
+  _hBlurTimeout;
+
+  /**
+   * The element ref of the tooltip's trigger button.
+   * @type {React.RefObject<Element>}
+   * @private
+   */
+  _triggerRef = React.createRef();
+
+  componentDidMount() {
+    const correctAlignment = this._getNewAlignment();
+    this._updateAlignment(correctAlignment);
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    const { onClose, langDir } = this.props;
+    // eslint-disable-next-line react/destructuring-assignment
+    if (!this.state.open && prevState.open && onClose) {
+      onClose();
+    }
+
+    if (prevProps.langDir !== langDir) {
+      const correctAlignment = this._getNewAlignment();
+      this._updateAlignment(correctAlignment);
+    }
+  }
+
+  static getDerivedStateFromProps({ open, tooltipAlignment }, state) {
+    const { prevOpen } = state;
+    return prevOpen === open
+      ? null
+      : {
+          open,
+          prevOpen: open,
+          tooltipAlignment,
+        };
+  }
+
+  componentWillUnmount() {
+    /* istanbul ignore if */
```

Comment:
Why do we ignore this if statement? Is this supposed to ignore the non existent else?

Reply:
@davidicus Thank you for commenting on this! It look like a bug, I will fix it.